test/app_profiler/profile_id_test.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module AppProfiler
+  class ProfileIdTest < TestCase
+    test "ProfileId memoization" do
+      ProfileId.reset
+
+      SecureRandom.expects(:hex).once.returns("profile-id")
```

Comment:
```suggestion
```
Don't need to mock for this test. Part of this test would be to actually call the secure random call because we mock it in most other places. 😄

Reply:
good feedback 👍 -- better than testing it via expectations